probe-rs/src/probe/cmsisdap/mod.rs

```diff
@@ -81,25 +81,41 @@ impl std::fmt::Debug for CmsisDap {
 }
 
 impl CmsisDap {
-    pub fn new_from_device(device: CmsisDapDevice) -> Self {
+    pub fn new_from_device(mut device: CmsisDapDevice) -> Result<Self, DebugProbeError> {
         // Discard anything left in buffer, as otherwise
         // we'll get out of sync between requests and responses.
         device.drain();
 
-        Self {
+        // Determine and set the report size. This method attempts
```

Comment:
I do not understand where it actually tries multiple report sizes, because from what I can see it just queries it from the probe. What am  I missing?

Reply:
I've removed this outdated comment.